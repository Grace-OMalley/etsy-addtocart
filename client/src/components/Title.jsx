import React from 'react';

const Title = (props) => {
  let starArray = [];
  for (let i = 0; i < 5; i++) {
    if (i < props.item.reviewAvg) {
      starArray.push(<svg key={i} className="solidstar" xmlns="http://www.w3.org/2000/svg" viewBox="3 3 18 18" aria-hidden="true" focusable="false"><path d="M20.83,9.15l-6-.52L12.46,3.08h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4Z"></path></svg>);
    } else {
      starArray.push(<svg key={i} className="hollowstar" xmlns="http://www.w3.org/2000/svg" viewBox="3 3 18 18" aria-hidden="true" focusable="false"><path d="M12,6.47l1.21,2.84.41,1,1.05.09,3.07.27-2.32,2-.8.69.24,1,.69,3L12.9,15.79l-.9-.53-.9.53L8.45,17.38l.69-3,.24-1-.8-.69-2.32-2,3.07-.27,1.05-.09.41-1L12,6.47m.46-3.39h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4-.28-.88-6-.52L12.46,3.08Z"></path></svg>);
    }
  }
  if (props.item.numberOfSales && props.item.brandName) {
    let count = 0;
    let numberSales = [];
    let numArray = props.item.numberOfSales.toString().split('');
    for (var i = numArray.length - 1; i >= 0; i--) {
      count++;
      numberSales.unshift(numArray[i]);
      if (count === 3) {
        numberSales.unshift(',');
        count = 0;
      }
    }
    let sales = numberSales.join('');
    return (
      <div>
        <h4 className="title">{props.item.brandName}</h4>
        <span className="salestotal">{sales} sales | </span>
        <span className="reviewstars">{starArray}</span>
        <br />
        <br />
        <h1>{props.item.itemName}</h1>
      </div>
    );
  } else {
    return (
      <div>Loading...</div>
    );
  }
};

export default Title;