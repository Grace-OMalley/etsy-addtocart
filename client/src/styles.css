body {
  background-color: white;
}
h1, h4 {
  margin: 0px;
  font-weight: normal;
}
.outerDiv {
  /* border: 1px solid black;
  border-radius: 5px; */
  float: right;
  width: 35%;
  margin-right: 30px;
}
.title {
  margin: 0px;
  font-weight: normal;
}
.title:hover {
  margin: 0px;
  text-decoration: underline;
}
.price {
  display: flex;
  align-items: center;
  justify-content: space-between;
  font-weight: normal;
}
.buttonbox, .specbox, .quantbox {
  text-align: center;
}
.button {
  border: 1px solid black;
  background-color: black;
  color: white;
  border-radius: 25px;
  width: 100%;
  height: 48px;
  transition: box-shadow 0.2s cubic-bezier(0.175, 0.885, 0.32, 1.275);
}
.button:hover {
  border: 1px solid black;
  background-color: black;
  color: white;
  border-radius: 25px;
  width: 100%;
  height: 48px;
  box-shadow: 0px 0px 0px 2px black;
  border-radius: 25px;
  transition: box-shadow 0.2s cubic-bezier(0.175, 0.885, 0.32, 1.275);
}
.sizetext, .quanttext{
  font-size: 12px;
  margin: 6px 0px 6px 0px;
}
.detailtext{
  font-size: 12px;
  margin: 0px 6px 0px 0px;
}
.desctext {
  font-size: 12px;
  margin: 0px 6px 0px 0px;
  background: linear-gradient(to top, #eee, #333);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}
.shiptext, .carttext {
  font-size: 12px;
  font-weight: normal;
  margin: 0px;
}
.img {
  height: 30px;
  width: 32px;
  padding-right: 10px;
}
.shipbox, .cartbox {
  display: flex;
  align-items: center;
  width: 100%;
}
.desc {
  width: 100%;
}
.descheader, .detailheader {
  display: flex;
  align-items: center;
  justify-content: space-between;
  transition: box-shadow 0.2s cubic-bezier(0.175, 0.885, 0.32, 1.275);
  border-radius: 25px;
}
.descheader:hover, .detailheader:hover {
  box-shadow: 0px 0px 0px 12px rgba(128, 128, 128, 0.119);
  border-radius: 25px;
  background-color: rgba(128, 128, 128, 0.119);
  transition: box-shadow 0.2s cubic-bezier(0.175, 0.885, 0.32, 1.275);
}
.star {
  width: 12px;
  height: 12px;
}
.specs, .quant {
  width: 100%;
  height: 48px;
  border-radius: 6px;
  border: 1px solid rgba(128, 128, 128, 0.233);
  box-shadow: 0px 0px 5px 4px rgba(128, 128, 128, 0.084);
  transition: box-shadow 0.6s cubic-bezier(0.175, 0.885, 0.32, 1.275);
}
.specs:hover, .quant:hover {
  width: 100%;
  height: 48px;
  border-radius: 6px;
  border: 1px solid rgba(128, 128, 128, 0.687);
  box-shadow: 0px 0px 5px 4px rgba(128, 128, 128, 0.185);
  transition: box-shadow 0.3s cubic-bezier(0.175, 0.885, 0.32, 1.275);
}
.learnmorebox {
  width: 100%;
  text-align: center;
}
.learnmore {
  font-size: 12px;
  background-color: white;
  border: white;
  border-radius: 25px;
  transition: box-shadow 0.3s cubic-bezier(0.175, 0.885, 0.32, 1.275);
}
.learnmore:hover {
  box-shadow: 0px 0px 0px 10px rgba(128, 128, 128, 0.119);
  border-radius: 25px;
  background-color: rgba(128, 128, 128, 0.119);
  transition: box-shadow 0.2s cubic-bezier(0.175, 0.885, 0.32, 1.275);
}
.uparrow {
  width: 12px;
  height: 6px;
  margin-right: 6px;
}
