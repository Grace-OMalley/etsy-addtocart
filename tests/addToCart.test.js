import TestRenderer from 'react-test-renderer';
import React from 'react';
import renderer from 'react-test-renderer';
import Title from '../client/src/components/Title.jsx';
import styles from '../client/src/styles.css';

const item = {
  specifications: [
    [ 'Extra Small' ],
    [ 'Small' ],
    [ 'Medium' ],
    [ 'Large' ],
    [ 'Extra Large' ]
  ],
  // _id: 6088edca4912fe3bada3375f,
  brandName: 'Jones - Morar',
  numberOfSales: 53156,
  reviewAvg: 3,
  itemName: 'Incredible Concrete Chicken',
  price: 297,
  inventoryCount: 11656,
  shippingHandlingTime: 1620796106543,
  itemDetails: 'Metal',
  itemDescription: 'The beautiful range of Apple NaturalÃ© that has an exciting mix of natural ingredients. With the Goodness of 100% Natural Ingredients',
  __v: 0
};

test('First Jest-React Test', () => {
  const component = renderer.create(
    <Title item={item}/>
  );
  let tree = component.toJSON();
  expect(component).toMatchSnapshot();
});