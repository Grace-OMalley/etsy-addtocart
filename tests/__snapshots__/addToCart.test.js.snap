// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First Jest-React Test 1`] = `
<div>
  <h4
    className="title"
  >
    Jones - Morar
  </h4>
  <span>
    53,156
     sales | 
  </span>
  <svg
    aria-hidden="true"
    className="star"
    focusable="false"
    viewBox="3 3 18 18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.83,9.15l-6-.52L12.46,3.08h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4Z"
    />
  </svg>
  <svg
    aria-hidden="true"
    className="star"
    focusable="false"
    viewBox="3 3 18 18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.83,9.15l-6-.52L12.46,3.08h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4Z"
    />
  </svg>
  <svg
    aria-hidden="true"
    className="star"
    focusable="false"
    viewBox="3 3 18 18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.83,9.15l-6-.52L12.46,3.08h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4Z"
    />
  </svg>
  <svg
    aria-hidden="true"
    className="star"
    focusable="false"
    viewBox="3 3 18 18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12,6.47l1.21,2.84.41,1,1.05.09,3.07.27-2.32,2-.8.69.24,1,.69,3L12.9,15.79l-.9-.53-.9.53L8.45,17.38l.69-3,.24-1-.8-.69-2.32-2,3.07-.27,1.05-.09.41-1L12,6.47m.46-3.39h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4-.28-.88-6-.52L12.46,3.08Z"
    />
  </svg>
  <svg
    aria-hidden="true"
    className="star"
    focusable="false"
    viewBox="3 3 18 18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12,6.47l1.21,2.84.41,1,1.05.09,3.07.27-2.32,2-.8.69.24,1,.69,3L12.9,15.79l-.9-.53-.9.53L8.45,17.38l.69-3,.24-1-.8-.69-2.32-2,3.07-.27,1.05-.09.41-1L12,6.47m.46-3.39h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4-.28-.88-6-.52L12.46,3.08Z"
    />
  </svg>
  <br />
  <br />
  <h1>
    Incredible Concrete Chicken
  </h1>
</div>
`;
